Mathlib/Analysis/LocallyConvex/WithSeminorms.lean

```diff
@@ -151,6 +151,31 @@ theorem basisSets_smul (U) (hU : U ∈ p.basisSets) :
   refine Set.Subset.trans (ball_smul_ball (s.sup p) √r √r) ?_
   rw [hU, Real.mul_self_sqrt (le_of_lt hr)]
 
+/-- If a family of seminorms is continuous, then their basis sets are neighborhoods of zero. -/
+lemma basisSets_mem_nhds {𝕜 E ι : Type*} [NormedField 𝕜]
+    [AddCommGroup E] [Module 𝕜 E] [TopologicalSpace E] (p : SeminormFamily 𝕜 E ι)
+    (hp : ∀ i, Continuous (p i)) (U : Set E) (hU : U ∈ p.basisSets) : U ∈ 𝓝 (0 : E) := by
+  obtain ⟨s, r, hr, rfl⟩ := p.basisSets_iff.mp hU
+  clear hU
+  refine Seminorm.ball_mem_nhds ?_ hr
+  classical
+  induction s using Finset.induction_on with
+  | empty => simpa using continuous_zero
+  | insert a s _ hs =>
+    simp only [Finset.sup_insert, coe_sup]
+    exact Continuous.max (hp a) hs
```

Comment:
I don't understand the reason for this move, could you explain?

Reply:
Originally I split the `FilterBasis` section into two sections, one where the scalars are seminormed rings and one for normed fields. I've now merged the results back into one `FilterBasis` section in their original order, but using `R` for seminormed rings and `𝕜` for normed rings.